### PR TITLE
[CFM-487-fix] Read-only group actions.

### PR DESCRIPTION
**Pull request by anvmn**
https://issuetracker.amplexor.com/jira/browse/CFM-487

### CODE REVIEW

File: project/profiles/capacity4more/modules/c4m/user/c4m_user/actions/c4m_user_contact_multiple.action.inc [hunk 1]
```diff
@@ -12,14 +12,23 @@
  * Declares action for Site Admin, Group Admins and Group Owner.
  */
 function c4m_user_contact_multiple_action_info() {
-  $group = c4m_og_current_group();
   // If not within OG context, don't declare action.
-  if (!$group) {
-    return array();
+  if (!$group = c4m_og_current_group()) {
+    return [];
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user/actions/c4m_user_contact_multiple.action.inc]: let's stick to the project convention and use `array()`

**Comment by anvmn** [comment]
**Wiki at Archived group:**
selection_391
selection_390

**No QP at archived group:**
selection_392

**No contact multiple at archived group:**
selection_394

**No CTA block at archived group:**
selection_393

File: project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_wiki/includes/c4m_features_og_wiki.theme.inc [hunk 2]
```diff
@@ -29,7 +28,7 @@ function theme_c4m_features_og_wiki_wiki(array $variables) {
 
     // Add wiki page button when user has permissions to add a wiki page.
     if ($group = og_context()) {
-      if (og_user_access('node', $group['gid'], 'administer group', $user)) {
+      if (node_access('create', 'wiki_page')) {
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_wiki/includes/c4m_features_og_wiki.theme.inc]: @anvmn 
Isn't this too generic change ?

File: project/profiles/capacity4more/modules/c4m/restful/c4m_restful_quick_post/c4m_restful_quick_post.module [hunk 3]
```diff
@@ -271,8 +272,9 @@ function c4m_restful_quick_post_form($group_id) {
     }
   }
 
-  // If all resources are turned off, disable the QP form.
-  if (empty($show_resources)) {
+  // If all resources are turned off, or user is not allowed to create a
+  // discussion, disable the QP form.
+  if (empty($show_resources) || !node_access('create', 'discussion')) {
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/restful/c4m_restful_quick_post/c4m_restful_quick_post.module]: @anvmn 
this one also seems to be too generic,
did you test this for SA, GO, GA and GM separately ?

File: project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_wiki/includes/c4m_features_og_wiki.theme.inc (hunk 2 above)
Comment (anvmn) [inline project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_wiki/includes/c4m_features_og_wiki.theme.inc]: This translates to 'current user can create a wiki page'.
As long as node_access() works properly, this should be fine.

File: project/profiles/capacity4more/modules/c4m/restful/c4m_restful_quick_post/c4m_restful_quick_post.module (hunk 3 above)
Comment (anvmn) [inline project/profiles/capacity4more/modules/c4m/restful/c4m_restful_quick_post/c4m_restful_quick_post.module]: For SA, GO and GM.

Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/restful/c4m_restful_quick_post/c4m_restful_quick_post.module]: Please confirm GA is fine as well

Comment (anvmn) [inline project/profiles/capacity4more/modules/c4m/restful/c4m_restful_quick_post/c4m_restful_quick_post.module]: Confirmed...

**Comment by ordavidil** [comment]
@anvmn 
is this ready for merge ?

**Comment by anvmn** [comment]
Yes it is

**Pull request merged**